### PR TITLE
[ENH] Add Weight Normalization Keras Wrapper

### PR DESCRIPTION
**Pull request by aadya940**
Related to #1714.

Implements the weight normalization algorithm as a wrapper rather than a layer.

### CODE REVIEW

**Comment by aadya940** [comment]
@hadifawaz1999

File: aeon/utils/tests/test_weightnorm.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+"""Tests for the Weight Normalization layer."""
+
+import os
+
+from aeon.utils.validation._dependencies import _check_soft_dependencies
+
+if _check_soft_dependencies(["tensorflow"], severity="none"):
```
Comment (MatthewMiddlehurst) [inline aeon/utils/tests/test_weightnorm.py:7]: I would import tensorflow inside the test rather than do this. There are a few tests with soft dependencies which conditionally skip using a decorator, I can find and link one if you need.

**Review by MatthewMiddlehurst — COMMENTED** [review]
This was the deprecated layer right? If there's any reused code I would link to the old package and acknowledge that even if it's gone. If it's all original, then no problem.

File: aeon/utils/tests/test_weightnorm.py (hunk 1 above)
Comment (aadya940) [reply aeon/utils/tests/test_weightnorm.py:7]: I'll just use `pytest.mark.skipif`.

**Comment by aadya940** [comment]
> This was the deprecated layer right? If there's any reused code I would link to the old package and acknowledge that even if it's gone. If it's all original, then no problem.

The implementation is original, hence no need to link to the old package.

**Comment by MatthewMiddlehurst** [comment]
Fine by me then. Will give it a bit more for @hadifawaz1999 to have a check over and put it in if theres nothing.

**Review by hadifawaz1999 — DISMISSED** [review]
Great Addition !! Do you think we can use it for the dilated cnn in deep clustering ?

**Comment by aadya940** [comment]
> Great Addition !! Do you think we can use it for the dilated cnn in deep clustering ?

@hadifawaz1999  Yes it should, since the online difference is a couple of zeros in the kernel. Added it in the test.

**Review by MatthewMiddlehurst — APPROVED** [review]

**Review by hadifawaz1999 — APPROVED** [review]
